pkg/kdump/kdump.js

```diff
@@ -0,0 +1,116 @@
+ */
+
+var cockpit = require("cockpit");
+
+var React = require("react");
+
+var kdumpView = require("./kdump-view.jsx");
+var kdumpClient = require("./kdump-client.es6");
+
+var initStore = function(rootElement) {
+    var dataStore = { };
+    dataStore.domRootElement = rootElement;
+
+    dataStore.kdumpClient = new kdumpClient.KdumpClient();
+
+    dataStore.applySettings = function(settings) {
+        var dfd = cockpit.defer();
+        dataStore.kdumpClient.testWriteLocation(settings)
+            .done(function() {
+                dataStore.kdumpClient.writeSettings(settings)
+                    .done(dfd.resolve)
+                    .fail(dfd.reject);
+            })
+            .fail(dfd.reject);
+        return dfd.promise();
+    };
+
+    // whether we're actively trying to change the state
+    dataStore.stateChanging = false;
+    function setServiceState(desiredState) {
+        if (dataStore.stateChanging) {
+            console.log("already trying to change state");
+            return;
+        }
+        var promise = desiredState ? dataStore.kdumpClient.ensureOn() : dataStore.kdumpClient.ensureOff();
+        dataStore.stateChanging = true;
+        dataStore.render();
+        promise.always(function() {
+            dataStore.stateChanging = false;
+            dataStore.render();
+        });
+    }
+    var render = function() {
+        React.render(React.createElement(kdumpView.page, {
+            kdumpActive: false,
+            onSetServiceState: setServiceState,
+            stateChanging: dataStore.stateChanging,
+            reservedMemory: dataStore.kdumpMemory,
+            kdumpStatus: dataStore.kdumpStatus,
+            onApplySettings: dataStore.applySettings,
+            onCrashKernel: dataStore.kdumpClient.crashKernel,
+        }), rootElement);
+    };
+    dataStore.render = render;
+
+    // read memory reserved for kdump from system
+    dataStore.kdumpMemory = undefined;
+    // HACK cockpit.file() can't be used for /sys yet, since those files can't be mapped
+    // https://github.com/cockpit-project/cockpit/issues/5597
+    // cockpit.file("/sys/kernel/kexec_crash_size").read()
+    cockpit.spawn(["cat", "/sys/kernel/kexec_crash_size"])
+        .done(function(content) {
+            var value = parseInt(content, 10);
+            if (!isNaN(value)) {
+                // if it's only a number, guess from the size what units we should use
+                // https://access.redhat.com/solutions/59432 states limit to be 896M and the auto at 768M max
+                // default unit is M
+                if (value >= 1000000)
+                    dataStore.kdumpMemory = cockpit.format_bytes(value);
+                else if (value >= 1000)
+                    dataStore.kdumpMemory = cockpit.format_bytes(value*1024);
+                else
+                    dataStore.kdumpMemory = cockpit.format_bytes(value*1024*1024);
+            } else {
+                dataStore.kdumpMemory = content.trim();
```

Comment:
Which values can be `kexec_crash_size`? I thought that there's always a value in bytes? Do we need the guessing the size then?

Note that `parseInt("256M", 10)` would also return `256`.